doc/dev/design/ui_goals.rst

```diff
@@ -0,0 +1,54 @@
+Ceph Dashboard Design Goals
+===========================
+
+.. note:: this document is intended to provide a focal point for discussing the overall design principles for mgr/dashboard
+
+Introduction
+############
+
+Most distributed storage architectures are inherently complex, and can present a management challenge to Operations teams who are
+typically stretched across multiple product and platform disciplines. In general terms, the complexity of any solution can have a 
+direct bearing on the operational costs incurred to manage it. The answer is simple...make it simple :)
+
+This document is intended to highlight Ceph Dashboard design goals which may help to  
+
+* reduce complexity
+* increase productivity
+* improve time-to-value
+
+ 
+Understanding the Persona of the Target User
+############################################
+
+Ceph has historically been administered from the CLI. The CLI has always and will always offer the richest, most flexible way to 
+install and manage a Ceph cluster. Administrators who require and demand this level of control are unlikely to adopt a UI for anything
+more than a technical curiousity.
+
+The relevance of the UI is therefore more critical for a new SysAdmin, where it can help technology adoption and reduce the operational friction
+that is normally experienced when implementing a new solution.
+
+Understanding the target user persona is therefore a fundamental first step in design. Attempting to design a UI that meets the requirements of a 'seasoned'
+Ceph Administrator or Developer, and a relatively new SysAdmin is unlikely to satisfy either user group.
```

Comment:
Agree. we need to have a "main target", and in my opinion this is the relatively new SysAdmin. 

But this does not mean not to provide the right tools to execute complex tasks.
For example:
We can provide a wizard to create osds in all available devices using the collocated schema
but
We must provide also the tool to create non collocated osds to improve performance using in the best way possible all storage devices.

Reply:
Yep - I guess what I'm alluding to is to minimize the practice of cli -> gui implementation. A gui full of ceph primitives is not going to help sysadmins pick up ceph - if anything it just preserves the view that Ceph is overly complicated, and difficult to manage.

Agree that in your OSD management example both implementation paths should be available as different workflows - that's the key thing though...implement as a workflow, not as a succession of separate tasks that the admin has to learn for themself.

Reply:
> A gui full of ceph primitives is not going to help sysadmins pick up ceph - if anything it just preserves the view that Ceph is overly complicated, and difficult to manage.

Fully agreed, and we've always said that this is just the first iteration. Next up now is taking all these "raw" building blocks that we've created and put them together in workflows that abstract the complexity away from inexperienced users.